library/ssl_tls13_server.c

```diff
@@ -1101,9 +1108,10 @@ static int ssl_write_server_pre_shared_key_ext( mbedtls_ssl_context *ssl,
 
     *olen = 0;
 
-    /* Are we using an external PSK? */
-    if( ssl->conf->psk == NULL || ssl->conf->psk_identity == NULL ||
-        ssl->conf->psk_identity_len == 0 || ssl->conf->psk_len == 0 )
+    /* Are we using any PSK at all? */
```

Comment:
Can we use `mbedtls_ssl_get_psk()` here?

Reply:
We could but we'd have to provide `mbedtls_ssl_get_psk()` with two non-null variables. Would that be acceptable?
If not, we've got several solutions to this problem in general:
+ Keep writing specific checks like this one
+ Add a function or change `mbedtls_ssl_get_psk()` to check the existence of a generic PSK (static or dynamic) in the  style of `ssl_conf_has_static_psk()`
  + Considering PSK checks are situation-dependent, it would require a complete check to accommodate all cases, so I don't think it makes sense

Reply:
I agree that it's not optimal, but in this case I'd prefer to allocate dummy variables on the stack to hold PSK pointer and PSK length.

Reply:
Independently, you can patch `mbedtls_ssl_get_psk()` to accept `NULL` parameters and thereby solely indicate whether a PSK is present or not. But I think that shouldn't hold off the PR.